tests/event.test.ts

```diff
@@ -0,0 +1,225 @@
+import { validate } from 'class-validator';
+import { plainToClass } from 'class-transformer';
+import { DatabaseConnection, EventFactory, FeedbackFactory, FileFactory, PortalState, UserFactory } from './data';
+import { ControllerFactory } from './controllers';
+import { UserAccessType } from '../types';
+import { Config } from '../config';
+import { StorageUtils } from './utils';
+import { SubmitEventFeedbackRequest } from '../api/validators/EventControllerRequests';
+
+beforeAll(async () => {
+  await DatabaseConnection.connect();
+});
+
+beforeEach(async () => {
+  await DatabaseConnection.clear();
+});
+
+afterAll(async () => {
+  await DatabaseConnection.clear();
+  await DatabaseConnection.close();
+});
+
+describe('event CRUD operations', () => {
+  test('events from the past, future, and all time can be pulled', async () => {
+    const conn = await DatabaseConnection.get();
+    const pastEvent = EventFactory.fakePastEvent();
+    const ongoingEvent = EventFactory.fakeOngoingEvent();
+    const futureEvent = EventFactory.fakeFutureEvent();
+    const user = UserFactory.fake();
+
+    await new PortalState()
+      .createEvents([pastEvent, ongoingEvent, futureEvent])
+      .createUsers([user])
+      .write();
+
+    const eventController = ControllerFactory.event(conn);
+
+    const pastEventsResponse = await eventController.getPastEvents({}, user);
```

Comment:
How come? I personally find it clearer to have all the assertions at the bottom so that it's easier to see what exactly is being asserted instead having assertions being tied within controller calls

Reply:
In general, it makes more sense to me to immediately assert instead of collecting all the responses so tests can fail faster and more obviously, e.g. if `getFutureEvents` throws then we fail and don't need to unnecessarily call `getAllEvents`. Also it isn't the case in this test but one call might influence the return value of another, so we should assert ASAP and make sure everything's correct at all times.

Reply:
Neither reason's particularly applicable here but I think it's still worth sticking to.